### PR TITLE
Introducing a template for identity schema

### PR DESCRIPTION
**Pull request by mrsabath**
**Pull Request check list**

- [ ] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

**Description of change**

**Which issue this PR fixes**

### CODE REVIEW

File: support/k8s/k8s-workload-registrar/config.go [hunk 1]
```diff
@@ -85,6 +87,15 @@ func (c *CommonMode) ParseConfig(hclConfig string) error {
 	if c.DisabledNamespaces == nil {
 		c.DisabledNamespaces = defaultDisabledNamespaces()
 	}
+	if c.IdentityTemplate == "" && c.Context != nil {
+		return errs.New("context defined without identity_template")
+	}
+	// IdentityTemplate represents the format after the trust domain and as such, it must not begin with spiffe://, // or /
+	if strings.HasPrefix(c.IdentityTemplate, "spiffe://") ||
+		strings.HasPrefix(c.IdentityTemplate, "//") ||
+		strings.HasPrefix(c.IdentityTemplate, "/") {
```
Comment (lumjjb) [inline support/k8s/k8s-workload-registrar/config.go:96]: I think this covers the above condition

File: support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go [hunk 2]
```diff
@@ -183,10 +201,43 @@ func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) string {
...
+		r.c.Log.WithError(err).Errorf("Error parsing the template %v", tpl)
+		log.Fatal(err)
+	}
+
+	// Create the IdentityMaps struct, with maps, one for Pod and one for Context:
+	podInfo := map[string]string{}
+	podInfo[PodServiceAccountIDLabel] = pod.Spec.ServiceAccountName
+	podInfo[NamespaceIDLabel] = pod.Namespace
+	podInfo[PodNameIDLabel] = pod.Name
+	podInfo[PodUIDLabel] = string(pod.UID)
```
Comment (lumjjb) [inline support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go]: how bout the following pattern:

```
podInfo := map[string]string{
  "PodServiceAccountIDLabel": pod.Spec.ServiceAccountName,
  "NamespaceIDLabel": pod.Namespace,
}
```

File: support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller_test.go [hunk 3]
```diff
@@ -66,18 +71,54 @@ func (s *PodControllerTestSuite) TestPodLabel() {
...
+			identityTemplate: "ns/{{.Pod.namespace}}/sa/{{.Pod.service_account}}/podName/{{.Pod.pod_name}}",
+			uid:              "012",
+			first:            fmt.Sprintf("ns/%s/sa/%s/podName/%s", PodNamespace, PodServiceAccount, PodName),
+			second:           fmt.Sprintf("ns/%s/sa/%s/podName/%s", PodNamespace, PodServiceAccount, PodName),
+		},
+		{
+			// Testing provided identity template, corresponding to a default format:
+			identityTemplate: "ns/{{.Pod.namespace}}/sa/{{.Pod.service_account}}",
+			first:            "ns/" + PodNamespace + "/sa/serviceAccount",
+			second:           "ns/" + PodNamespace + "/sa/serviceAccount",
```
Comment (lumjjb) [inline support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller_test.go]: shouldn't "serviceAccount" be using the variable as well?

File: support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller_test.go [hunk 4]
```diff
@@ -86,20 +127,23 @@ func (s *PodControllerTestSuite) TestPodLabel() {
 				Namespace:   PodNamespace,
 				Labels:      map[string]string{"spiffe": test.first},
 				Annotations: map[string]string{"spiffe": test.first},
+				//	UID:         types.UID(test.uid),
```
Comment (lumjjb) [inline support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller_test.go]: nit: dead code

File: support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller_test.go [hunk 5]
```diff
@@ -86,20 +127,23 @@ func (s *PodControllerTestSuite) TestPodLabel() {
...
+				ServiceAccountName: "serviceAccount",
 			},
 		}
 		err := s.k8sClient.Create(s.ctx, &pod)
 		s.Require().NoError(err)
 		s.reconcile(p)
+
 		labelSelector := labels.Set(map[string]string{
-			"podUid": string(pod.ObjectMeta.UID),
+			// "podUid": string(pod.ObjectMeta.UID),
```
Comment (lumjjb) [inline support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller_test.go]: nit: dead code

**Review by lumjjb — COMMENTED** [review]
Nice - minor nits and small suggestions! Looks nicely shaping up!

**Comment by mrsabath** [comment]
Creating PR against the upstream repo

**Pull request closed without merging**